python/tk_nuke/menu_generation.py

```diff
@@ -21,291 +18,665 @@
 import nukescripts.openurl
 import nukescripts
 
+from PySide import QtGui
+
+# -----------------------------------------------------------------------------
 
 class MenuGenerator(object):
     """
-    Menu generation functionality for Nuke
+    The factory base class for Nuke based menu generators. Since Nuke
+    now provides multiple modes (Nuke, Nuke Studio, and Hiero), we are
+    required to supply menu generation logic for each. Instantiating
+    this class directly will provide the correct implementation for the
+    currently running Nuke mode.
     """
+    def __new__(cls, *args, **kwargs):
```

Comment:
Yeah, I think that's fine. This was the first thing that I merged and I originally had grand plans of making it all transparent and not needing if/else statements everywhere, but in the end I went with the simpler approach with the rest of the engine. I'll clean this up and make it more straightforward.